opc/resource_orchestrated_instance.go

```diff
@@ -0,0 +1,240 @@
+package opc
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/hashicorp/go-oracle-terraform/client"
+	"github.com/hashicorp/go-oracle-terraform/compute"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+)
+
+func resourceOPCOrchestratedInstance() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceOPCOrchestratedInstanceCreate,
+		Read:   resourceOPCOrchestratedInstanceRead,
+		Delete: resourceOPCOrchestratedInstanceDelete,
+		Update: resourceOPCOrchestratedInstanceUpdate,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(20 * time.Minute),
+			Update: schema.DefaultTimeout(20 * time.Minute),
+			Delete: schema.DefaultTimeout(20 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"description": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"desired_state": {
+				Type:     schema.TypeString,
+				Required: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					"active",
+					"inactive",
+					"suspend",
+				}, true),
+			},
+			"tags": tagsOptionalSchema(),
+
+			"instance": orchestrationInstanceSchema(),
+
+			"version": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceOPCOrchestratedInstanceCreate(d *schema.ResourceData, meta interface{}) error {
+	log.Printf("[DEBUG] Resource state: %#v", d.State())
```

Comment:
Is this line printing anything? State would be pretty much empty at this point, save configuration.

Reply:
I believe there is benefit here in cases where the orchestrated instance is dependent on other resources.